server/src/test/java/org/elasticsearch/cluster/coordination/UnsafeBootstrapMasterIT.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.cluster.coordination;
+
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.cli.MockTerminal;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.Manifest;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.discovery.DiscoverySettings;
+import org.elasticsearch.discovery.zen.ElectMasterService;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.env.NodeEnvironment;
+import org.elasticsearch.env.NodeMetaData;
+import org.elasticsearch.env.TestEnvironment;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.InternalTestCluster;
+import org.elasticsearch.test.junit.annotations.TestLogging;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0, autoMinMasterNodes = false)
+@TestLogging("_root:DEBUG,org.elasticsearch.cluster.service:TRACE,org.elasticsearch.discovery.zen:TRACE")
+public class UnsafeBootstrapMasterIT extends ESIntegTestCase {
+
+    private int bootstrapNodeId;
```

Comment:
This seems to be used more as a flag indicating whether or not to bootstrap the cluster, and a default of `do bootstrap the cluster` makes more sense to me.

Reply:
Well, in fact, it's not a flag. It's the sequence number of the started node, which is used to bootstrap the cluster. I'm using `autoMinMasterNodes=false`, which means I don't want auto-bootstrap. If we set `bootstrapNodeId` to something different than -1, what value to choose? Shall we bootstrap the cluster after the first node is started or the third node is started? So by default, the behaviour is the same as `autoMinMasterNodes=false`, but the tests that need the cluster bootstrap and understand when this bootstrap should be performed can set this field to some positive value.
Please note, this code is copy pasted in several test classes already, I plan to have a follow-up PR that refactors this method to `ESIntegTestCase` so that tests with `autoMinMasterNodes=false` can have an easy way to perform the bootstrap.

Reply:
Ok